application/src/main/java/org/togetherjava/tjbot/features/javamail/RSSHandlerRoutine.java

```diff
@@ -0,0 +1,412 @@
+        if (textChannel == null) {
+            logger.warn("Tried to sendRss, got empty response (channel {} not found)",
+                    feedConfig.targetChannelPattern());
+            return;
+        }
+
+        List<Item> rssItems = fetchRSSItemsFromURL(feedConfig.url());
+        if (rssItems.isEmpty()) {
+            return;
+        }
+
+        // We only get the first RSS item from the list since we are
+        // assuming that the whole RSS feed uses the same format
+        if (!isValidDateFormat(rssItems.getFirst(), feedConfig)) {
+            logger.warn("Could not find valid date format for RSS feed {}", feedConfig.url());
+            return;
+        }
+
+        RssFeedRecord rssFeedRecord = getRssFeedRecordFromDatabase(feedConfig);
+        Optional<ZonedDateTime> lastSavedDate = getLastSavedDateFromDatabaseRecord(rssFeedRecord,
+                feedConfig.dateFormatterPattern());
+
+        ZonedDateTime lastPostedDate =
+                getLatestPostDateFromItems(rssItems, feedConfig.dateFormatterPattern());
+        updateLastDateToDatabase(feedConfig, rssFeedRecord, lastPostedDate);
+
+        if (lastSavedDate.isEmpty()) {
+            return;
+        }
+
+        final Predicate<Item> shouldItemBePosted = item -> {
+            ZonedDateTime itemPubDate =
+                    getDateTimeFromItem(item, feedConfig.dateFormatterPattern());
+            return itemPubDate.isAfter(lastSavedDate.get());
+        };
+
+        rssItems.reversed()
+            .stream()
+            .filter(shouldItemBePosted)
+            .forEach(item -> postItem(textChannel, item, feedConfig));
+    }
+
+    /**
+     * Retrieves an RSS feed record from the database based on the provided RSS feed configuration.
+     *
+     * @param feedConfig the RSS feed configuration to retrieve the record for
+     * @return the RSS feed record retrieved from the database
+     */
+    private RssFeedRecord getRssFeedRecordFromDatabase(RSSFeed feedConfig) {
+        return database.read(context -> context.selectFrom(RSS_FEED)
+            .where(RSS_FEED.URL.eq(feedConfig.url()))
+            .limit(1)
+            .fetchAny());
+    }
+
+    /**
+     * Retrieves the last saved date from the database record associated with the given RSS feed
+     * record.
+     * <p>
+     * If the RSS feed record is null, returns an empty {@link Optional}.
+     *
+     * @param rssRecord the RSS feed record to retrieve the last saved date from
+     * @param dateFormatterPattern the pattern used to parse the date from the database record
+     * @return An {@link Optional} containing the last saved date if it could be retrieved and
+     *         parsed successfully, otherwise an empty {@link Optional}
+     */
+    private Optional<ZonedDateTime> getLastSavedDateFromDatabaseRecord(RssFeedRecord rssRecord,
+            String dateFormatterPattern) {
+        if (rssRecord == null) {
```

Comment:
Depending of what you do from my comment in getRssFeedRecordFromDatabase, improve this, it's usually not a good idea to allow null params to then return null/optional.empty.